src/api/unified/symbol_manager.cpp

```diff
@@ -133,10 +135,11 @@ LibHandle openDynLibrary(const af_backend bknd_idx) {
     typedef af_err (*func)(int*);
 
     LibHandle retVal = nullptr;
-    for (size_t i = 0; i < extent<decltype(paths)>::value; i++) {
-        AF_TRACE("Attempting: {}", paths[i]);
-        if ((retVal = loadLibrary(join_path(paths[i], bkndLibName).c_str()))) {
-            AF_TRACE("Found: {}", join_path(paths[i], bkndLibName));
+    for (size_t i = 0; i < extent<decltype(pathPrefixes)>::value; i++) {
+        AF_TRACE("Attempting: {}", pathPrefixes[i]);
+        if ((retVal = loadLibrary(
+                 join_path(pathPrefixes[i], bkndLibName).c_str()))) {
+            AF_TRACE("Found: {}", join_path(pathPrefixes[i], bkndLibName));
 
             func count_func =
                 (func)getFunctionPointer(retVal, "af_get_device_count");
```

Comment:
Another failure condition is when the library is found but there are no devices associated with the library. This happens if you have an AMD GPU and you try to load the CUDA backend.

Reply:
I don't understand do you want us to throw error in that case ? because we are already logging the device count in Line 149.

One logic where the library is still usable would be to cycle through each backend in the order of preference and switch to the first backend that we come across with at least one device. Even then, it wouldn't be error unless all backends has zero devices.

I will handle device selection logic in different PR, don't want to mix that with this logging related changes.